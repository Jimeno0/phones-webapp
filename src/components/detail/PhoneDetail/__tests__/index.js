import { PhoneDetail } from '../'
import React from 'react'
import { render } from '@testing-library/react'

const phoneData = {
  id: '0',
  name: 'iPhone 7',
  manufacturer: 'Apple',
  description: 'iPhone 7 dramatically improves the most important aspects of the iPhone experience. It introduces advanced new camera systems. The best performance and battery life ever in an iPhone. Immersive stereo speakers. The brightest, most colorful iPhone display. Splash and water resistance*. And it looks every bit as powerful as it is. This is iPhone 7.',
  color: 'black',
  price: 769,
  imageFileName: 'http://localhost:8080/images/IPhone_7.png',
  screen: '4,7 inch IPS',
  processor: 'A10 Fusion',
  ram: 2
}

describe('Phone detail', () => {
  test('it renders', () => {
    const { container } = render(<PhoneDetail />)
    expect(container).toBeTruthy()
  })

  test('Snapshot', () => {
    const { container } = render(<PhoneDetail />)
    expect(container).toMatchSnapshot()
  })

  test('renders details info elements', () => {
    const { getByText } = render(<PhoneDetail {...phoneData} />)
    const Title = getByText(phoneData.name)
    const Ram = getByText(`${phoneData.ram} Gb`)
    const Processor = getByText(phoneData.processor)
    const Screen = getByText(phoneData.screen)
    const Manufacturer = getByText(`By ${phoneData.manufacturer}`)
    const Price = getByText(`$${phoneData.price}`)
    const Description = getByText(phoneData.description)

    expect(Title).toBeTruthy()
    expect(Ram).toBeTruthy()
    expect(Processor).toBeTruthy()
    expect(Screen).toBeTruthy()
    expect(Manufacturer).toBeTruthy()
    expect(Price).toBeTruthy()
    expect(Description).toBeTruthy()
  })
})
