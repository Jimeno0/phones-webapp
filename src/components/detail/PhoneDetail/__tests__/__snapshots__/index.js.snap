// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phone detail Snapshot 1`] = `
<div>
  <div
    class="sc-AxhUy jGheux"
    data-testid="phone-detail"
  >
    <img
      alt="phone"
      class="sc-AxheI iPBBu"
    />
    <div
      class="sc-AxhUy hjgyGc"
    >
      <div
        class="sc-AxiKw ebEjkl"
        size="48"
      />
      <h4
        class="sc-AxirZ hysMXl"
      />
      <div
        class="sc-AxiKw bqZeR"
        size="40"
      />
      <div
        class="sc-AxhUy fCzkFS"
      >
        <div
          class="sc-AxhCb fkmski"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.251.068a.5.5 0 01.227.58L9.677 6.5H13a.5.5 0 01.364.843l-8 8.5a.5.5 0 01-.842-.49L6.323 9.5H3a.5.5 0 01-.364-.843l8-8.5a.5.5 0 01.615-.09z"
              fill-rule="evenodd"
            />
          </svg>
          <div
            class="sc-AxiKw iFeIoW"
            size="16"
          />
          <p
            class="sc-AxjAm iuOAYh"
            color="textDesc"
          >
             Gb
          </p>
        </div>
        <div
          class="sc-AxhCb fkmski"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M228.844 32.22v114.218h17.687V32.218h-17.686zm-108.25.624c-15.507 0-28.094 12.586-28.094 28.093C92.5 76.444 105.087 89 120.594 89c12.655 0 23.34-8.372 26.844-19.875h44.937v77.313h17.688v-95H147.03c-3.888-10.837-14.262-18.593-26.436-18.593zm193.25 0c-15.507 0-28.063 12.586-28.063 28.093 0 12.124 7.677 22.45 18.44 26.376v59.124h17.655V87.844c11.596-3.452 20.063-14.193 20.063-26.906 0-15.508-12.587-28.094-28.094-28.094zM266.124 92.5v53.938h17.657V92.5h-17.655zm188.532 4.03c-15.507 0-28.094 12.588-28.094 28.095 0 13.083 8.948 24.074 21.063 27.188v27.468h-92.938v17.657h110.624v-46.342c10.223-4.192 17.407-14.233 17.407-25.97 0-15.507-12.557-28.094-28.064-28.094zM30.187 123.657v17.688H96.75v55.594h62.814V179.28h-45.126v-55.624h-84.25zm147.032 40.47v159.718h159.81v-159.72H177.22zm17.56 15.655h17.657v78.595l32.407 32.406h75.28v17.658H237.5l-2.594-2.594-10.75-10.75c-1.033 7.385-7.36 13.062-15.03 13.062-8.392 0-15.19-6.796-15.19-15.187 0-7.682 5.696-13.98 13.095-15l-9.655-9.658-2.594-2.593V179.78zm54.94.157h17.686v55.313h52.53l.002 17.688H249.72v-73zM53.124 217.375V307.344c-11.49 3.512-19.844 14.198-19.844 26.844 0 15.505 12.557 28.093 28.064 28.093s28.093-12.587 28.093-28.092c0-12.195-7.79-22.564-18.656-26.438v-72.72h88.782v-17.655H53.124zm301.563 0v17.656h53.968v-17.655h-53.97zm99.968 21.97c-10.898 0-20.342 6.21-25 15.28h-74.97l.002 17.688H427c2.325 13.168 13.824 23.187 27.656 23.187 15.507 0 28.063-12.588 28.063-28.094 0-15.507-12.557-28.062-28.064-28.062zm-349.062 15.28v17.688h53.97v-17.688h-53.97zm17.156 36.47v84.217c-11.498 3.513-19.875 14.2-19.875 26.844 0 15.506 12.587 28.094 28.094 28.094 15.506 0 28.06-12.588 28.06-28.094 0-12.194-7.766-22.564-18.624-26.437v-66.94h19.156v-17.686H122.75zm231.938 0v17.686h45.156v95.283c-11.323 3.624-19.53 14.26-19.53 26.78-.002 15.506 12.585 28.063 28.092 28.063 15.507 0 28.063-12.557 28.063-28.062 0-12.32-7.935-22.778-18.97-26.563V291.095h-62.814zM192.375 341.53v54.033h17.688V341.53h-17.688zm36.47 0v86.564c-11.013 3.794-18.94 14.233-18.94 26.53 0 15.506 12.588 28.095 28.095 28.095s28.063-12.59 28.063-28.095c0-12.53-8.203-23.14-19.532-26.75V341.53h-17.686zm37.28 0v54.033h17.688l-.032-54.032h-17.655zm38.094 0v140.064h17.655V341.53H304.22z"
            />
          </svg>
          <div
            class="sc-AxiKw iFeIoW"
            size="16"
          />
          <p
            class="sc-AxjAm iuOAYh"
            color="textDesc"
          />
        </div>
        <div
          class="sc-AxhCb dkTRRO"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M1.5 1a.5.5 0 00-.5.5v4a.5.5 0 01-1 0v-4A1.5 1.5 0 011.5 0h4a.5.5 0 010 1h-4zM10 .5a.5.5 0 01.5-.5h4A1.5 1.5 0 0116 1.5v4a.5.5 0 01-1 0v-4a.5.5 0 00-.5-.5h-4a.5.5 0 01-.5-.5zM.5 10a.5.5 0 01.5.5v4a.5.5 0 00.5.5h4a.5.5 0 010 1h-4A1.5 1.5 0 010 14.5v-4a.5.5 0 01.5-.5zm15 0a.5.5 0 01.5.5v4a1.5 1.5 0 01-1.5 1.5h-4a.5.5 0 010-1h4a.5.5 0 00.5-.5v-4a.5.5 0 01.5-.5z"
              fill-rule="evenodd"
            />
          </svg>
          <div
            class="sc-AxiKw iFeIoW"
            size="16"
          />
          <p
            class="sc-AxjAm iuOAYh"
            color="textDesc"
          />
        </div>
      </div>
      <div
        class="sc-AxiKw iFeIoW"
        size="16"
      />
      <p
        class="sc-AxjAm epggMG"
        color="textDesc"
      >
        By 
      </p>
      <div
        class="sc-AxiKw iFeIoW"
        size="16"
      />
      <div
        class="sc-AxhUy chQsXs"
      >
        <p
          class="sc-AxjAm epggMG"
          color="textDesc"
        >
          Color:
        </p>
        <div
          class="sc-AxiKw kyrQuB"
          size="16"
        />
        <div
          class="sc-AxgMl bZMToy"
        />
      </div>
      <div
        class="sc-AxiKw jsbzMH"
        size="24"
      />
      <p
        class="sc-AxjAm jMEEmv"
        color="primary"
        size="20"
      >
        $
      </p>
      <div
        class="sc-AxiKw jsbzMH"
        size="24"
      />
      <p
        class="sc-AxjAm epggMG"
        color="textDesc"
      />
    </div>
  </div>
</div>
`;
